parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -869,20 +869,19 @@ oras_login_with_kubelet_identity() {
     echo "successfully logged in to acr '$acr_url' with identity token"
 }
 
-# Localdns corefile is created only when localdns profile has state enabled.
+# localdns corefile is created only when localdns profile has state enabled.
 # This should match with 'path' defined in parts/linux/cloud-init/nodecustomdata.yml.
 LOCALDNS_CORE_FILE="/opt/azure/containers/localdns/localdns.corefile"
-# This function is called in cse_main.sh. 
+# This function is called from cse_config.sh.
 # It checks if the localdns corefile exists and is not empty.
 # If the corefile exists and is not empty, it returns 0 - localdns should be enabled.
-# If the corefile does not exist or is empty, 
-# it returns ERR_LOCALDNS_COREFILE_NOTFOUND exit code - localdns should not be enabled.
-shouldEnableLocaldns() {
+# If the corefile does not exist or is empty, it returns 1 - localdns should not be enabled.
+shouldEnableLocalDNS() {
     if [ ! -f "${LOCALDNS_CORE_FILE}" ] || [ ! -s "${LOCALDNS_CORE_FILE}" ]; then
```

Comment:
why not just add a bash variable in cse_cmd.sh containing the value of `ShouldEnableLocalDNS`, since that's what's used to determine whether the corefile gets created

Reply:
for now this is fine, though when we refactor towards not using customdata for scriptless I'd opt for making this change as well

Reply:
enablement should _only_ be done via CSE, avoid using go templates anywhere outside cse_cmd.sh to make things compatible out of the box with other bootstrappers